cl/phase1/network/services/attestation_service.go

```diff
@@ -239,8 +241,12 @@ func (s *attestationService) ProcessMessage(ctx context.Context, subnet *uint64,
 		},
 	}
 
+	if att.ImmediateProcess {
+		return s.batchSignatureVerifier.ImmediateVerification(aggregateVerificationData)
```

Comment:
I'm not sure we want to return the error here. So, if we have incorrect/invalid signature we already ban the peer in `ImmediateVerification` and by returning error we will cause the peer to be banned again. It goes the same for publishing gossip data if signature verification succeeds. If we return `nil` here we will be publishing the gossip data again that we already did in `ImmediateVerification`

So, that's why I returned `ErrIgnore` from `ProcessMessage` so that no peer banning or gossip publishing would happen because we already do that in here. So maybe you should simply run `ImmediateVerification` and return `ErrIgnore`

Reply:
this is fine, `ImmediateVerification` seems to be only done when we receive messages from the API anyway so this is probably correct